taqueria-vscode-extension/src/lib/helpers.ts

```diff
@@ -263,12 +266,57 @@ export class VsCodeHelper {
 			}
 		});
 		await this.updateCommandStates();
+
+		// When there workspace changes, sent all tracked events
+		this.vscode.workspace.onDidChangeWorkspaceFolders(() => this.sendEvents());
+
+		this.vscode.window.onDidChangeWindowState(() => this.sendEvents());
+
+		// Track an event indicating the vscode extension has been loaded
+		await this.trackEvent('taq_vscode_loaded', {
+			ext_version: this.context.extension.packageJSON.version,
+			vscode_version: this.vscode.version,
+		});
+
+		await this.sendEvents();
 	}
 
 	get i18n() {
 		return this.i18;
 	}
 
+	async getAnalytics() {
+		if (!this.analytics) {
+			const taqVersion = (await this.getTaqVersion()) || 'unknown';
+			const taqBuild = (await this.getTaqBuild()) || 'unknown';
+
+			this.analytics = Analytics.inject({
+				taqVersion: taqVersion.trim(),
+				taqBuild: taqBuild.trim(),
+				isCICD: false,
```

Comment:
If we implement integration/e2e tests for TVsCE, this might need to be changed

Reply:
Is there something in particular that you're concerned about Ali that warrants adjustment?